openedx/core/djangoapps/models/course_details.py

```diff
@@ -210,7 +210,7 @@ def update_from_json(cls, course_key, jsondict, user):  # pylint: disable=too-ma
             converted = date.from_json(jsondict['start_date'])
         else:
             converted = None
-        if converted != descriptor.start:
+        if converted != descriptor.start and converted:
```

Comment:
I think it’s better to move the `converted` check to the beginning of the if statement, like this:
`if converted and converted != descriptor.start:`

Reply:
agree